components/tableNodes/OfferTable.component.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from "react";
+import Button from "@mui/material/Button";
+import Chip from "@mui/material/Chip";
+import Paper from "@mui/material/Paper";
+import Table from "@mui/material/Table";
+import Box from "@mui/material/Box";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import useSWR from "swr";
+import Loading from "../genericView/Loading.component";
+import { JSX } from "@babel/types";
+import theme from "../../theme/DarkTheme";
+import { fetcher, selectedOffer } from "../../utils/AppUtils";
+import { makeStyles } from "@mui/styles";
+
+type OfferTableProps = {
+  show: (visible: boolean, message: string) => void;
+};
+
+const useStyles = makeStyles({
+  root: {
+    width: "100%",
+  },
+  container: {
+    maxHeight: "100vh",
+    maxWidth: "100vw",
+  },
+});
+
+export function OfferTable({ show }: OfferTableProps): JSX.Element {
+  const { data, error } = useSWR("/api/offer", fetcher);
+
+  const classes = useStyles();
+  if (error) {
+    show(true, "Error: " + error);
+    return <></>;
+  }
+  if (!data) return <Loading />;
+  if (data.offer.length == 0) {
+    show(true, "No offers currently");
+    return <></>;
+  }
+  return (
+    <Box mt={theme.spacing(1)} mb={theme.spacing(2)} className={classes.root}>
+      <TableContainer component={Paper} className={classes.container}>
+        <Table stickyHeader aria-label="Node that shows the list of offers">
+          <TableHead>
+            <TableRow>
+              <TableCell>Select</TableCell>
+              <TableCell>Offer Id</TableCell>
+              <TableCell>Active</TableCell>
+              <TableCell>Use</TableCell>
```

Comment:
```
              <TableCell>label</TableCell>
               <TableCell>Active</TableCell>
               <TableCell>Select</TableCell>
```

we change the order, and also what we add inside the table

Reply:
P.S: all method that you have in `listoffers`

Reply:
`label` is optional though, right?

Reply:
yes, label is optional but we can add if there is no label we add `"------"`

>and no need of offer_id

For what do we need it?